kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -53,8 +55,30 @@ class SparkProcessBuilder(
     Paths.get(sparkHome, "bin", SPARK_SUBMIT_FILE).toFile.getCanonicalPath
   }
 
+  initialize
+
   override def mainClass: String = "org.apache.kyuubi.engine.spark.SparkSQLEngine"
 
+  /**
+   * Add or override spark.master if KUBERNETES_SERVICE_HOST and
+   * KUBERNETES_SERVICE_PORT are defined
+   */
+  protected def initialize(): Unit = {
+    try {
+      if (conf.get(KUBERNETES_MASTER_OVERRIDE)) {
```

Comment:
how about try auto configure it iff `spark.master` is absent?

Reply:
I agree with your view and complete `spark.master` only in case of  its absence.